### ISSUE TITLE
[BUG]: incorrect precision, recall, and f1-score for object dtype target

### ISSUE THREAD

**Issue by kwhkim** [pycaret/pycaret#3900]
### pycaret version checks

- [X] I have checked that this issue has not already been reported [here](https://github.com/pycaret/pycaret/issues).

- [X] I have confirmed this bug exists on the [latest version](https://github.com/pycaret/pycaret/releases) of pycaret.

- [X] I have confirmed this bug exists on the master branch of pycaret (pip install -U git+https://github.com/pycaret/pycaret.git@master).

### Issue Description

predict_model() prints wrong accuracy, precision, and recall score when target is of object dtype.

### Reproducible Example

```python
# +
# tutorial from https://nbviewer.org/github/pycaret/pycaret/blob/master/tutorials/Tutorial%20-%20Binary%20Classification.ipynb
# -

import numpy as np
import pandas as pd

# +
from sklearn.metrics import accuracy_score, precision_score, recall_score, f1_score, cohen_kappa_score, matthews_corrcoef

def metrics_binary(y_true, y_pred, pos_label=1):
    return pd.DataFrame({'value':[accuracy_score(y_true, y_pred),
                                  precision_score(y_true, y_pred, pos_label=pos_label),
                                  recall_score(y_true, y_pred, pos_label=pos_label),
                                  f1_score(y_true, y_pred, pos_label=pos_label),
                                  cohen_kappa_score(y_true, y_pred),
                                  matthews_corrcoef(y_true, y_pred)]},
                         index = ['acc', 'precision', 'recall',
                                  'f1', 'kappa', 'mcc'])
# -

from sklearn.model_selection import train_test_split

# +

# loading sample dataset from pycaret dataset module
from pycaret.datasets import get_data
data = get_data('diabetes')

# -

data.columns



# ### target 0 or 1

X_train, X_test, y_train, y_test = train_test_split(data.drop('Class variable', axis=1),
                                                    data['Class variable'],
                                                    test_size=0.2)



dat_train = pd.concat([X_train, y_train], axis=1)
dat_test = pd.concat([X_test, y_test], axis=1)

# +


# import pycaret classification and init setup
from pycaret.classification import *
s = setup(dat_train, target = 'Class variable', 
      test_data = dat_test,
          session_id = 123)


# -

gbc = s.create_model('gbc')

pred = s.predict_model(gbc)

# +
#pred['prediction_label']
# ...
# 695    1
# 451    0
# Name: prediction_label, Length: 154, dtype: int64

# +
# y_test
# -



metrics_binary(y_test, pred['prediction_label']).T

# ### object target

data = get_data('diabetes')



data['Class variable'] = np.where(data['Class variable']==1,
                                  "Positive",
                                  "Negative")

data.head()

X_train, X_test, y_train, y_test = train_test_split(data.drop('Class variable', axis=1),
                                                    data['Class variable'],
                                                    test_size=0.2)
dat_train = pd.concat([X_train, y_train], axis=1)
dat_test = pd.concat([X_test, y_test], axis=1)

s2 = setup(dat_train, target = 'Class variable', 
      test_data = dat_test,
          session_id = 123)

gbc = s.create_model('gbc')

pred = s2.predict_model(gbc)

pd.crosstab(y_test, pred['prediction_label'])

metrics_binary(y_test, pred['prediction_label'], pos_label="Positive").T

```
```


### Expected Behavior

The first part of the example above comes from https://nbviewer.org/github/pycaret/pycaret/blob/master/tutorials/Tutorial%20-%20Binary%20Classification.ipynb. and it works fine.

![image](https://github.com/pycaret/pycaret/assets/4476168/738a1a43-6363-44f4-be4a-f54ab06a1fd3)


But when I change the target dtype to object, reports from `predict_model()` shows wrong precision, recall and f1 score which can be verified from the second part.


![image](https://github.com/pycaret/pycaret/assets/4476168/6a15247b-62ee-4e84-9d61-8e7c73bf6a63)


### Actual Results

```python-traceback
Wrong metrics(precision, recall, f1)
```

### Installed Versions

System:
    python: 3.9.18 | packaged by conda-forge | (main, Dec 23 2023, 16:33:10)  [GCC 12.3.0]
executable: /home/publishingkwon/miniforge3/envs/py39mpl36sk12pdp02pycaret240131/bin/python
   machine: Linux-5.15.0-1049-gcp-x86_64-with-glibc2.31

PyCaret required dependencies:
                 pip: 23.3.2
          setuptools: 69.0.3
             pycaret: 3.2.1
             IPython: 8.18.1
          ipywidgets: 8.1.1
                tqdm: 4.66.1
               numpy: 1.25.0
              pandas: 1.5.3
              jinja2: 3.1.3
               scipy: 1.10.1
              joblib: 1.3.2
             sklearn: 1.2.2
                pyod: 1.1.2
            imblearn: 0.12.0
   category_encoders: 2.6.3
            lightgbm: 4.3.0
               numba: 0.58.1
            requests: 2.31.0
          matplotlib: 3.6.1
          scikitplot: 0.3.7
         yellowbrick: 1.5
              plotly: 5.18.0
    plotly-resampler: Not installed
             kaleido: 0.2.1
           schemdraw: 0.15
         statsmodels: 0.14.1
              sktime: 0.21.1
               tbats: 1.1.3
            pmdarima: 2.0.3
              psutil: 5.9.8
          markupsafe: 2.1.4
             pickle5: Not installed
         cloudpickle: 3.0.0
         deprecation: 2.1.0
              xxhash: 3.4.1
           wurlitzer: 3.0.3

PyCaret optional dependencies:
                shap: 0.44.0
           interpret: 0.5.0
                umap: 0.5.5
     ydata_profiling: 0.0.dev0
  explainerdashboard: 0.4.5
             autoviz: Not installed
           fairlearn: 0.7.0
          deepchecks: Not installed
             xgboost: 2.0.3
            catboost: 1.2.2
              kmodes: 0.12.2
             mlxtend: 0.23.1
       statsforecast: 1.5.0
        tune_sklearn: 0.4.6
                 ray: 2.9.1
            hyperopt: 0.2.7
              optuna: 3.5.0
               skopt: 0.9.0
              mlflow: 2.10.0
              gradio: 4.16.0
             fastapi: 0.109.0
             uvicorn: 0.27.0.post1
              m2cgen: 0.10.0
           evidently: 0.2.6
               fugue: 0.8.7
           streamlit: Not installed
             prophet: Not installed

**Comment by kwhkim** [comment]
What could possibly lead to this bug??? I am curious...

**Comment by celestinoxp** [comment]
@kwhkim can you check with new pycaret 3.3.0 ? you can use `pip install --upgrade pycaret`
If the problem is resolved, can you close this issue, if not, can you provide more details?

**Comment by moezali1** [comment]
Closing as part of the 4.0 backlog cleanup — see #4177.

PyCaret 4.0 is a ground-up rewrite (OOP-only API, sklearn-native engine, Python 3.11+, sklearn ≥ 1.7) and almost all pre-4.0 issues either reference removed APIs (`setup()`, `plot_model`, `evaluate_model`, `interpret_model`, `automl`, the legacy time-series module, the Streamlit dashboard, …) or no longer reproduce on the new code path.

**If this still affects you in 4.0, please re-file as a new issue** with:

- the experiment constructor call (`Experiment(...)`),
- the operation that fails,
- the full traceback,
- your Python / sklearn / pycaret versions.

Resources for 4.0:

- Docs: https://pycaret.org/docs
- Migrate from 3.x: https://pycaret.org/docs/guides/migrating-from-3-x
- API reference: https://pycaret.org/reference
- FAQ: https://pycaret.org/docs/resources/faqs

Thanks for filing — and apologies for the long latency. The repo had drifted; we're starting clean.
